src/features/world/roomMachine.ts

```diff
@@ -164,116 +177,67 @@ export const roomMachine = createMachine<ChatContext, RoomEvent, RoomState>({
-          },
-          {
-            target: "ready",
-            actions: assign({
-              room: (_, event) => event.data.room,
-            }),
-          },
-        ],
         onError: {
           target: "error",
+          cond: (_, event) => !event.data.roo,
+          // Fire off an event, and let the game render player anyway
+        },
+        onDone: {
+          target: "ready",
+          actions: assign({
+            roomId: (_, event) => event.data.roomId,
+            rooms: (context, event) => {
+              return {
+                ...context.rooms,
+                [event.data.roomId]: event.data.room,
+              };
+            },
+          }),
         },
       },
     },
     ready: {
       on: {
         CHANGE_ROOM: {
-          target: "initialising",
-          actions: assign({
-            roomId: (_, event) => event.roomId,
-            messages: (_) => [],
-            players: (_) => ({}),
-          }),
+          target: "joinRoom",
         },
         SEND_CHAT_MESSAGE: {
           actions: (context, event) => {
-            context.room?.send(0, { text: event.text });
+            const room = context.rooms[context.roomId];
+            if (!room) return {};
+
+            room.send(0, { text: event.text });
           },
         },
         SEND_POSITION: {
           actions: (context, event) => {
-            context.room?.send(0, {
+            const room = context.rooms[context.roomId];
+            if (!room) return {};
+
+            room.send(0, {
               x: event.x,
               y: event.y,
             });
           },
         },
-        CHAT_MESSAGE_RECEIVED: {
-          actions: assign({
-            messages: (context, event) => {
-              return [
-                {
-                  text: event.text,
-                  sessionId: event.sessionId,
-                },
-                ...context.messages,
-              ];
-            },
-          }),
-        },
-        PLAYER_MOVED: {
-          actions: assign({
-            players: (context, event) => {
-              return {
-                ...context.players,
-                [event.sessionId]: {
-                  x: event.x,
-                  y: event.y,
-                },
-              };
-            },
-          }),
-        },
-        PLAYER_JOINED: {
-          actions: assign({
-            players: (context, event) => {
-              return {
-                ...context.players,
-                [event.sessionId]: {
-                  x: event.x,
-                  y: event.y,
-                  clothing: event.clothing,
-                },
-              };
-            },
-          }),
-        },
-        PLAYER_QUIT: {
-          actions: assign({
-            players: (context, event) => {
-              const newPlayers = context.players;
-
-              delete newPlayers[event.sessionId];
-
-              return newPlayers;
-            },
-          }),
-        },
```

Comment:
The context this was updating has been removed, so for now these events don't need processing directly by the machine.

Reply:
Love this!